### PR TITLE
Migrate System to inter-op

### LINKED ISSUE #20056: Migrate system to interop
**Description:**
$subject

### PR DESCRIPTION
**Pull request by kalaiyarasiganeshalingam**
## Purpose
Migrate config-api to inter-op

Fixes #20056

## Check List 
- [ ] Read the [Contributing Guide](https://github.com/ballerina-platform/ballerina-lang/blob/master/CONTRIBUTING.md)
- [ ] Required Balo version update
- [ ] Updated Change Log
- [ ] Checked Tooling Support (#<Issue Number>)
- [ ] Added necessary tests
  - [ ] Unit Tests
  - [ ] Spec Conformance Tests
  - [ ] Integration Tests
  - [ ] Ballerina By Example Tests
- [ ] Increased Test Coverage   
- [ ] Added necessary documentation  
  - [ ] API documentation 
  - [ ] Module documentation in Module.md files
  - [ ] Ballerina By Examples

### CODE REVIEW

**Comment by aashikam** [comment]
This should be Migrate system to inter-op isn't it?

File: stdlib/grpc/build.gradle [hunk 1]
```diff
@@ -73,6 +74,7 @@ dependencies {
     baloImplementation project(path: ':ballerina-crypto', configuration: 'baloImplementation')
     baloImplementation project(path: ':ballerina-time', configuration: 'baloImplementation')
     baloImplementation project(path: ':ballerina-http', configuration: 'baloImplementation')
+    baloImplementation project(path: ':ballerina-java', configuration: 'baloImplementation')
```
Comment (aashikam) [inline stdlib/grpc/build.gradle]: Why do we need this?

File: cli/ballerina-cli-module-pull/build.gradle [hunk 2]
```diff
@@ -64,10 +64,10 @@ dependencies {
     baloImplementation project(path: ':ballerina-stringutils', configuration: 'baloImplementation')
     baloImplementation project(path: ':ballerina-java', configuration: 'baloImplementation')
 
-    interopImports project(':ballerina-filepath')
```
Comment (aashikam) [inline cli/ballerina-cli-module-pull/build.gradle]: Why was this removed?

File: stdlib/streams/build.gradle [hunk 3]
```diff
@@ -70,7 +71,6 @@ dependencies {
     testCompile project(path: ':ballerina-test-common', configuration: 'tests')
     testCompile project(path: ':ballerina-test-utils', configuration: 'shadow')
     testCompile project(':ballerina-core')
-    testCompile project(':ballerina-filepath')
```
Comment (aashikam) [inline stdlib/streams/build.gradle]: Why was this removed?

File: stdlib/system/build.gradle [hunk 4]
```diff
@@ -39,6 +39,10 @@ dependencies {
     baloImplementation project(path: ':ballerina-time', configuration: 'baloImplementation')
     baloImplementation project(path: ':ballerina-lang:internal', configuration: 'baloImplementation')
     baloImplementation project(path: ':ballerina-io', configuration: 'baloImplementation')
+    baloImplementation project(path: ':ballerina-java', configuration: 'baloImplementation')
+
+    interopImports project(':ballerina-system')
```
Comment (aashikam) [inline stdlib/system/build.gradle]: This is the system module gradle so why do you need this `interopImports project(':ballerina-system')` here?

File: cli/ballerina-cli-module-pull/build.gradle (hunk 2 above)
Comment (kalaiyarasiganeshalingam) [reply cli/ballerina-cli-module-pull/build.gradle]: I have added. https://github.com/ballerina-platform/ballerina-lang/pull/20095/files#diff-69d9c615e318006f2090fba78b10148bR68

File: stdlib/system/src/main/ballerina/src/system/system.bal [hunk 5]
```diff
@@ -14,27 +14,56 @@
...
 # + name - Name of the environment variable
 # + return - Environment variable value if it exists, else an empty string
-public function getEnv(@untainted string name) returns string = external;
+public function getEnv(@untainted string name) returns string {
+    return covertToString(nativeGetEnv(java:fromString(name)));
+}
+
+function nativeGetEnv(handle key) returns handle = @java:Method {
+    name: "getEnv",
+    class: "org.ballerinalang.stdlib.system.nativeimpl.GetEnv"
```
Comment (ThisaruGuruge) [inline stdlib/system/src/main/ballerina/src/system/system.bal]: We can use the `java.lang.System.getEnv()` for this, no need to use another class here, since what the `GetEnv` class only does is to call that particular method.

It will return the string as a `handle` and we can convert it to ballerina `string`.

Handle the `nil` value properly.

You can see how to use `java.lang` methods in inter-op [here](https://github.com/ballerina-platform/ballerina-lang/blob/03ffe281f821bc43f8dfd79f1d80df0d45764004/stdlib/stringutils/src/main/ballerina/src/stringutils/natives.bal#L166).

Please check for similar cases.

File: stdlib/system/src/main/java/org/ballerinalang/stdlib/system/nativeimpl/Uuid.java [hunk 6]
```diff
@@ -18,27 +18,16 @@
...
-@BallerinaFunction(
-        orgName = "ballerina", packageName = "system",
-        functionName = "uuid",
-        returnType = {@ReturnType(type = TypeKind.STRING)},
-        isPublic = true
-)
 public class Uuid {
 
-    public static String uuid(Strand strand) {
+    public static String uuid() {
```
Comment (ThisaruGuruge) [inline stdlib/system/src/main/java/org/ballerinalang/stdlib/system/nativeimpl/Uuid.java]: We can get rid of this class as well.

File: cli/ballerina-cli-module-pull/build.gradle (hunk 2 above)
Comment (aashikam) [reply cli/ballerina-cli-module-pull/build.gradle]: Ack

File: stdlib/system/src/main/java/org/ballerinalang/stdlib/system/nativeimpl/Uuid.java (hunk 6 above)
Comment (kalaiyarasiganeshalingam) [reply stdlib/system/src/main/java/org/ballerinalang/stdlib/system/nativeimpl/Uuid.java]: Fixed

File: stdlib/system/src/main/ballerina/src/system/system.bal [hunk 7]
```diff
@@ -43,6 +78,24 @@ public function uuid() returns string = external;
...
+function nativeExec(handle command, map<string> env, string? dir, string[] args) returns Process|Error = @java:Method {
+    name: "exec",
+    class: "org.ballerinalang.stdlib.system.nativeimpl.Exec"
+} external;
+
+function covertToString(handle value) returns string {
+    string output = "";
+    var result = java:toString(value);
+    if (result is string) {
+        output = result;
```
Comment (ThisaruGuruge) [inline stdlib/system/src/main/ballerina/src/system/system.bal]: We can remove the variable `output` and return the result directly right ?

Comment (kalaiyarasiganeshalingam) [reply stdlib/system/src/main/ballerina/src/system/system.bal]: Fixed

File: stdlib/system/src/main/java/org/ballerinalang/stdlib/system/nativeimpl/GetEnv.java [hunk 8]
```diff
@@ -17,22 +17,16 @@
  */
```
Comment (ThisaruGuruge) [inline stdlib/system/src/main/java/org/ballerinalang/stdlib/system/nativeimpl/GetEnv.java]: We can delete this class, since we don't use it right ?

File: stdlib/system/src/main/ballerina/src/system/system.bal [hunk 9]
```diff
@@ -43,6 +77,24 @@ public function uuid() returns string = external;
...
+    } else {
+        error err = error("Error occured when converting the value to string.");
```
Comment (anupama-pathirage) [inline stdlib/system/src/main/ballerina/src/system/system.bal]: This can be reduced to one line. i.e. no need of `err` variable

File: stdlib/system/src/main/ballerina/src/system/system.bal (hunk 7 above)
Comment (ThisaruGuruge) [reply stdlib/system/src/main/ballerina/src/system/system.bal]: This isn't fixed right ?

Comment (ThisaruGuruge) [reply stdlib/system/src/main/ballerina/src/system/system.bal]: Still the `output` variable is there, right?

Comment (aashikam) [reply stdlib/system/src/main/ballerina/src/system/system.bal]: This was not fixed.

Comment (kalaiyarasiganeshalingam) [reply stdlib/system/src/main/ballerina/src/system/system.bal]: Fixed in https://github.com/ballerina-platform/ballerina-lang/pull/20113/
